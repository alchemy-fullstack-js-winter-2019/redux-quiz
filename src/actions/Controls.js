export const UPDATE_TITLE = 'UPDATE_TITLE';
export const updateTitle = (title) => {
    return {
        type: UPDATE_TITLE,
        payload: title
    };
};
export const UPDATE_SUBTITLE = 'UPDATE_SUBTITLE';
export const updateSubtitle = (subtitle) => {
    return {
        type: UPDATE_SUBTITLE,
        payload: subtitle
    };
};
