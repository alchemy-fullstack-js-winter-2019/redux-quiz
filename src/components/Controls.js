import React from 'react';
import PropTypes from 'prop-types';

function Controls({ onChange, topText, bottomText }) {
  return (
    <>
    <label>
        Top Text: 
      <input type="text" name="topText" value={topText} onChange={onChange}/>
    </label>
    <label>
        Bottom Text: 
      <input type="text" name="bottomText" value={bottomText} onChange={onChange}/>
    </label>
    </>
  );
}

Controls.propTypes = {
  onChange: PropTypes.func.isRequired,
  topText: PropTypes.string.isRequired,
  bottomText: PropTypes.string.isRequired
};

export default Controls;
