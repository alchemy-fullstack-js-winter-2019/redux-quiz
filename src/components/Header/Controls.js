import React from 'react';
import PropTypes from 'prop-types';


function Controls({ title, subtitle, onChange }) {
  return (
    <>
  Enter Title:
      <input type='text' name="title" value={title} onChange={onChange} />
  Enter Subtitle:
      <input type="text" name="subtitle" value={subtitle} onChange={onChange} />
    </>
  );
}

Controls.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired
};

export default Controls;
