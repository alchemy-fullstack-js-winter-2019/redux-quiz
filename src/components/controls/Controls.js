import React from 'react';
import PropTypes from 'prop-types';

function Controls({ title, subtitle, onChange }) {
  return (
    <>
      <input name='title' value={title} onChange={onChange} />
      <input name='subtitle' value={subtitle} onChange={onChange} />
    </>
  );
}

Controls.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export default Controls;
