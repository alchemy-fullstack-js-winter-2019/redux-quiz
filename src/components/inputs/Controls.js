import React from 'react';
import PropTypes from 'prop-types';

const Controls = ({ title, subtitle, handleChange }) => {
  return (
    < >
      <label htmlFor="title">
        Enter a title:<br/>
        <input name="title"
          value={title}
          onChange={handleChange}
        />
      </label>
      <br/>
      <label htmlFor="subtitle">
        Enter a subtitle:<br/>
        <input name="subtitle"
          value={subtitle}
          onChange={handleChange}
        />
      </label>
    </>
  );
};

Controls.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired
};

export default Controls;
