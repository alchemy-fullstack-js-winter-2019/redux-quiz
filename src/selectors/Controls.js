export const getTitle = state => {
    return state.Controls.title;
};

export const getSubtitle = state => {
    return state.Controls.subtitle;
}
;
