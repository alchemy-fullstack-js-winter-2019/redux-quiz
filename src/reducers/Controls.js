
import { UPDATE_SUBTITLE, UPDATE_TITLE } from '../actions/Controls';

const initalState = {
    title: '',
    subtitle: ''
};
export default function reducer(state = initalState, action) {
    switch(action.type) {
        case UPDATE_TITLE:
            return { ...state, title: action.payload };
        case UPDATE_SUBTITLE:
            return { ...state, subtitle: action.payload };
        default:
            return state;
    }
}
