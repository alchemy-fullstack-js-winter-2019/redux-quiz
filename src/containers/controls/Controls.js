import { connect } from 'react-redux';
import Controls from '../../components/controls/Controls';
import { getTitle, getSubtitle } from '../../selectors/Controls';
import { updateTitle, updateSubtitle } from '../../actions/Controls';

const mapStateToProps = state => ({
    title: getTitle(state),
    subtitle: getSubtitle(state)

});

const makeDispatchToProps = dispatch => ({
    onChange({ target }) {
        const factoryMethod = {
            title: updateTitle,
            subtitle: updateSubtitle
        };
        dispatch(factoryMethod[target.name](target.value));
    }
});

export default connect(mapStateToProps, makeDispatchToProps)(Controls);
