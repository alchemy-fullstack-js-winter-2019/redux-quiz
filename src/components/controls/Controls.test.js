import React from 'react';
import renderer from 'react-test-renderer';
import Controls from './Controls';

describe('Controls', () => {
  it('matched a snapshot', () => {
    const tree = renderer.create(
      <Controls />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

//add props

