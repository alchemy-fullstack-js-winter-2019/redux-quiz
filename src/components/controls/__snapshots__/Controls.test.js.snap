// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls matched a snapshot 1`] = `
Array [
  <input
    name="title"
  />,
  <input
    name="subtitle"
  />,
]
`;
